app/src/main/java/com/internship/move/MainViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package com.internship.move
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.lifecycle.ViewModel
+import com.internship.move.feature.authentification.register.RegisterFragment
+import com.internship.move.feature.map.MapFragment
+
+open class MainViewModel : ViewModel() {
```

Comment:
I see you chose to make this class open and extend it with empty bodies as you needed.
I would suggest changing the implementation such that each viewModel has only the methods it needs to work instead of extending this god class that handles everything.
This way we make sure changes aren't made from places they aren't supposed to be made (separation of concerns)